zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+                reply["channel"] = post_dict["channel"]
+                raw_messages.append(message_to_dict(reply))
+
+    def process_batch(lst: List[Dict[str, Any]]) -> None:
+        process_raw_message_batch(
+            realm_id=realm_id,
+            raw_messages=lst,
+            subscriber_map=subscriber_map,
+            user_id_mapper=user_id_mapper,
+            user_handler=user_handler,
+            get_recipient_id=get_recipient_id,
+            is_pm_data=is_pm_data,
+            output_dir=output_dir,
+        )
+
+    chunk_size = 1000
+
+    process_list_in_batches(
+        lst=raw_messages,
+        chunk_size=chunk_size,
+        process_batch=process_batch,
+    )
+
+def write_message_data(team_name: str,
+                       realm_id: int,
+                       post_data: List[Dict[str, Any]],
+                       zerver_recipient: List[ZerverFieldsT],
+                       subscriber_map: Dict[int, Set[int]],
+                       output_dir: str,
+                       masking_content: bool,
+                       stream_id_mapper: IdMapper,
+                       user_id_mapper: IdMapper,
+                       user_handler: UserHandler,
+                       username_to_user: Dict[str, Dict[str, Any]]) -> None:
+
+    stream_id_to_recipient_id = {
+        d['type_id']: d['id']
+        for d in zerver_recipient
+        if d['type'] == Recipient.STREAM
+    }
+
+    def get_stream_recipient_id(raw_message: ZerverFieldsT) -> int:
+        receiver_id = raw_message['receiver_id']
+        stream_id = stream_id_mapper.get(receiver_id)
+        recipient_id = stream_id_to_recipient_id[stream_id]
+        return recipient_id
+
+    process_posts(
+        team_name=team_name,
+        realm_id=realm_id,
+        post_data=post_data,
+        get_recipient_id=get_stream_recipient_id,
+        subscriber_map=subscriber_map,
+        output_dir=output_dir,
+        is_pm_data=False,
+        masking_content=masking_content,
+        user_id_mapper=user_id_mapper,
+        user_handler=user_handler,
+        username_to_user=username_to_user,
+    )
+
+def initialize_username_to_user_mapping(user_data_list: List[Dict[str, Any]],
+                                        username_to_user: Dict[str, Dict[str, Any]]) -> None:
+    for user in user_data_list:
+        username_to_user[user["username"]] = user
+
+def check_user_in_team(user: Dict[str, Any], team_name: str) -> bool:
+    for team in user["teams"]:
+        if team["name"] == team_name:
+            return True
+    return False
+
+def label_mirror_dummy_users(team_name: str, mattermost_data: Dict[str, List[Dict[str, Any]]],
+                             username_to_user: Dict[str, Dict[str, Any]]) -> None:
+    for post in mattermost_data["post"]:
+        if post["team"] == team_name:
+            user = username_to_user[post["user"]]
+            if not check_user_in_team(user, team_name):
+                user["is_mirror_dummy"] = True
```

Comment:
Probably this would be a good place to label which users are admins (obviously with a different function name)

Reply:
hmm. We are iterating only through users that has send at-least one message here. So all the admins wont be necessarily labelled here.

Reply:
Ahh, OK, makes sense.  Maybe worth a comment, since I could easily see someone thinking my proposal was a good refactor in the future